packages/opentelemetry-node/src/spanprocessor.ts

```diff
@@ -20,6 +20,32 @@ export const SENTRY_SPAN_PROCESSOR_MAP: Map<SentrySpan['spanId'], SentrySpan> =
  * the Sentry SDK.
  */
 export class SentrySpanProcessor implements OtelSpanProcessor {
+  public constructor() {
+    addGlobalEventProcessor(event => {
+      const otelSpan = trace.getActiveSpan();
+      if (!otelSpan) {
+        return event;
+      }
+
+      const otelSpanId = otelSpan.spanContext().spanId;
+      const sentrySpan = SENTRY_SPAN_PROCESSOR_MAP.get(otelSpanId);
+
+      if (!sentrySpan) {
+        return event;
+      }
+
+      // If event has already set `trace` context, use that one.
+      // This happens in the case of transaction events.
+      event.contexts = { trace: sentrySpan.getTraceContext(), ...event.contexts };
```

Comment:
l: Not sure, but maybe it is more efficient to mutate this? To avoid always creating a new object here? (Also, not sure if that is even possible 😅 ) E.g.:

```js
if (!event.context.trace) {
  event.context.trace = sentrySpan.getTraceContext();
}
```

Reply:
Typescript complains that `event.contexts` might not be defined. I don't think this is too much of a performance penalty and makes intent very clear on what is happening.

Ditto with tags below.

Reply:
👍